core/jazz_acl/components/casbin.js

```diff
@@ -98,19 +102,19 @@ async function addOrRemovePolicy(serviceId, config, action, policies) {
   let result = {};
   let conn;
   const objects = [`${serviceId}_manage`, `${serviceId}_code`, `${serviceId}_deploy`];
-  let totalPolices = 0;
+  let totalPolicies = 0;
   try {
     const getPolicies = await getFilteredPolicy(1, objects, config);
 
-    if (getPolicies) {
-      totalPolicies = getPolicies.length;
+    if (getPolicies) {//found previous policies to be removed
+      totalPolicies = getPolicies.filter(policy => policy.length > 0).length;
       let removeResult = [];
 
       if (totalPolicies) {
         conn = await dbConnection(config);
         const enforcer = await casbin.newEnforcer('./config/rbac_model.conf', conn);
 
-        // remove (incase of add remove first)
+        // remove (incase of add, remove first)
         if (action === 'remove' || action === 'add') {
           removeResult = await getPolicies.map(async policies => {
```

Comment:
if totalpolicies is zero, then removal shouldn't happen i guess.

Reply:
yes and it wont(see line 113)